datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -108,13 +108,9 @@ fn on_lr_is_preserved(plan: &LogicalPlan) -> Result<(bool, bool)> {
             JoinType::Left => Ok((false, true)),
             JoinType::Right => Ok((true, false)),
             JoinType::Full => Ok((false, false)),
-            JoinType::LeftSemi
-            | JoinType::LeftAnti
-            | JoinType::RightSemi
-            | JoinType::RightAnti => {
-                // filter_push_down does not yet support SEMI/ANTI joins with join conditions
-                Ok((false, false))
-            }
+            JoinType::LeftSemi | JoinType::RightSemi => Ok((true, true)),
```

Comment:
I thought a Semi join had the same semantics as Left join -- namely that the left/right side were preserved but the  others weren't

So I would expect  this to be something like


```suggestion
            JoinType::LeftSemi => Ok((false, true)),
            JoinType::RightSemi => Ok((true, false)),
```

Maybe I am mis understanding the semi join

Reply:
I played around with it in postgres:

```sql

postgres=# create table bar as values (2, 'a'), (3, 'b');
SELECT 2
postgres=# create table foo as values (1, 'X'), (2, 'Y'), (3, 'Z');
SELECT 3


postgres=# explain select * from foo SEMI JOIN bar USING (column1) where bar.column2 != 'b' ;
                               QUERY PLAN                                
-------------------------------------------------------------------------
 Merge Join  (cost=179.17..305.88 rows=8026 width=68)
   Merge Cond: (bar.column1 = semi.column1)
   ->  Sort  (cost=90.99..94.15 rows=1264 width=36)
         Sort Key: bar.column1
         ->  Seq Scan on bar  (cost=0.00..25.88 rows=1264 width=36)
               Filter: (column2 <> 'b'::text)
   ->  Sort  (cost=88.17..91.35 rows=1270 width=36)
         Sort Key: semi.column1
         ->  Seq Scan on foo semi  (cost=0.00..22.70 rows=1270 width=36)
(9 rows)

postgres=# select * from foo SEMI JOIN bar USING (column1);
 column1 | column2 | column2 
---------+---------+---------
       2 | Y       | a
       3 | Z       | b
(2 rows)

postgres=# select * from foo SEMI JOIN bar USING (column1) where bar.column2 != 'b' ;
 column1 | column2 | column2 
---------+---------+---------
       2 | Y       | a
(1 row)
```

And this this looks correct to me

Reply:
Update 🤦  the postgres `foo SEMI JOIN bar` is being interpreted as though `SEMI` is an alias, not a different join type